queue.go

```diff
@@ -511,3 +520,20 @@ func (q *Queue) receiveJob(ctx context.Context) *amqp.Delivery {
 		return &job
 	}
 }
+
+func ReadCfgFromEnv() (*url, error) {
+	url := &url{}
+	envSet, err := env.EnvironToEnvSet(os.Environ())
+	if err != nil {
+		return nil, err
+	}
+	err = env.Unmarshal(envSet, url)
+	if err != nil {
+		return nil, err
+	}
+	return url, nil
+}
+
+func (url *url) urlString() string {
```

Comment:
please move it to be close to the `type url struct`

Reply:
done